homeassistant/components/spotify/__init__.py

```diff
@@ -76,15 +79,12 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
         return True
 
     if CONF_CLIENT_ID in config[DOMAIN]:
-        config_flow.SpotifyFlowHandler.async_register_implementation(
+        await async_import_client_credential(
```

Comment:
We should deprecate the existing YAML configuration and log a warning if its still present (after import)

Reply:
OK, I was thinking we'd need a few other things before that e.g. frontend bump, updated user instructions page about how the application credentials interface works before that.  Will circle back.

Reply:
(I was assuming i'd log warnings and mark deprecated after completing those, which maybe you're saying isn't right)

Reply:
I've added the warning, and we can discuss what other user facing documentation we need in place before merging, or merge and work on that in parallel if it's fine.

Reply:
Your latest change looks about right, we generally wrap the configuration schema with cv.deprecated as well (so the domain key triggers a warning too, e.g., configuration checks).

Reply:
👍🏼 this now logs two warnings on startup.

Reply:
Yeah, it does indeed... I  don't think it's harmful, as the bigger log message gives more details to the user about what has happened.